server/pulp/server/managers/repo/unit_association.py

```diff
@@ -414,15 +414,21 @@ def calculate_associated_type_ids(source_repo_id, associated_units):
 
 
 def create_transfer_units(associate_units, associated_unit_type_ids):
-    type_defs = {}
-    for def_id in associated_unit_type_ids:
-        type_def = types_db.type_definition(def_id)
-        type_defs[def_id] = type_def
+    unit_key_fields = {}
+    for unit_type_id in associated_unit_type_ids:
+        type_def = types_db.type_definition(unit_type_id)
+        if type_def is not None:
```

Comment:
I considered that and initially went that way, but here's why I changed my mind. As-is, the code that gets the type_def is near the code that uses it. If that doesn't work, we move on to the code that gets the model and then uses it. Ignoring control statements, it can be reduced to:

get A
use A
get B
use B

I think this a more friendly and readable layout than the alternative, but it's definitely a subtlety.